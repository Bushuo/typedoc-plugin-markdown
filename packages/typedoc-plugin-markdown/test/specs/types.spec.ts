import * as Handlebars from 'handlebars';

import { TestApp } from '../test-app';

describe(`Types:`, () => {
  let testApp: TestApp;

  beforeAll(async () => {
    testApp = new TestApp(['types.ts']);
    await testApp.bootstrap();
  });

  test(`should compile 'array' type'`, () => {
    expect(
      TestApp.compileHelper(
        Handlebars.helpers.type,
        testApp.findReflection('arrayType').type,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile 'stringLiteral' type'`, () => {
    expect(
      TestApp.compileHelper(
        Handlebars.helpers.type,
        testApp.findReflection('stringLiteralType').type,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile 'union' of string literals types'`, () => {
    expect(
      TestApp.compileHelper(
        Handlebars.helpers.type,
        testApp.findReflection('unionType').type,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile 'union' of literal declarations`, () => {
    expect(
      TestApp.compileHelper(
        Handlebars.helpers.type,
        testApp.findReflection('unionTypeWithSymbolsDeclarations').type,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile intrinsic type'`, () => {
    expect(
      Handlebars.helpers.type.call(testApp.findReflection('stringType').type),
    ).toMatchSnapshot();
  });

  test(`should compile collapsed 'literal' type'`, () => {
    expect(
      Handlebars.helpers.type.call(
        testApp.findReflection('literalType').type,
        'all',
      ),
    ).toMatchSnapshot();
  });

  test(`should compile expanded 'literal' type'`, () => {
    expect(
      Handlebars.helpers.type.call(testApp.findReflection('literalType').type),
    ).toMatchSnapshot();
  });

  test(`should compile collapsed 'objectLiteralType' type'`, () => {
    expect(
      Handlebars.helpers.type.call(
        testApp.findReflection('objectLiteralType'),
        'object',
      ),
    ).toMatchSnapshot();
  });

  test(`should compile expanded 'objectLiteralType' type'`, () => {
    expect(
      Handlebars.helpers.type.call(testApp.findReflection('objectLiteralType')),
    ).toMatchSnapshot();
  });

  test(`should compile 'tuple' type'`, () => {
    expect(
      Handlebars.helpers.type.call(testApp.findReflection('tupleType').type),
    ).toMatchSnapshot();
  });

  test(`should compile 'intersection' type'`, () => {
    expect(
      Handlebars.helpers.type.call(
        testApp.findReflection('intersectionType').type,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile collapsed 'function' type '`, () => {
    expect(
      Handlebars.helpers.type.call(
        testApp.findReflection('functionReflectionType').type,
        'function',
      ),
    ).toMatchSnapshot();
  });

  test(`should compile expanded 'function' type '`, () => {
    expect(
      Handlebars.helpers.type.call(
        testApp.findReflection('functionReflectionType').type,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile 'typeOperator' type '`, () => {
    expect(
      Handlebars.helpers.type.call(
        testApp.findReflection('typeOperatorType').type,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile unionType with object literal type '`, () => {
    expect(
      Handlebars.helpers.type.call(
        testApp.findReflection('objectLiteralUnionType').type,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile conditional type '`, () => {
    expect(
      Handlebars.helpers.type.call(
        testApp.findReflection('ConditionalType').type,
      ),
    ).toMatchSnapshot();
  });

  test(`should resolve external refs'`, () => {
    expect(
      Handlebars.helpers.type.call(
        testApp.findReflection('externalReference').type,
      ),
    ).toMatchSnapshot();
  });

  test(`should resolve external refs with type params'`, () => {
    expect(
      Handlebars.helpers.type.call(
        testApp.findReflection('externalReferenceInsideTypeParams').type,
      ),
    ).toMatchSnapshot();
  });
});
