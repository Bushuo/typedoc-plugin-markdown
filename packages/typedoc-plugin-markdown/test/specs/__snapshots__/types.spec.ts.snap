// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Types: should compile 'array' type' 1`] = `
"string[]
"
`;

exports[`Types: should compile 'intersection' type' 1`] = `"[IntersectionClassA](intersectionclassa.md) & [IntersectionClassB](intersectionclassb.md)"`;

exports[`Types: should compile 'stringLiteral' type' 1`] = `
"\\"blue\\"
"
`;

exports[`Types: should compile 'tuple' type' 1`] = `"[string, number]"`;

exports[`Types: should compile 'typeOperator' type ' 1`] = `"unique symbol"`;

exports[`Types: should compile 'union' of string literals types' 1`] = `
"\\"ease-in\\" \\\\| \\"ease-out\\" \\\\| \\"ease-in-out\\"
"
`;

exports[`Types: should compile collapsed 'function' type ' 1`] = `"function"`;

exports[`Types: should compile collapsed 'literal' type' 1`] = `"object"`;

exports[`Types: should compile collapsed 'objectLiteralType' type' 1`] = `"object"`;

<<<<<<< HEAD
exports[`Types: should compile conditional type ' 1`] = `"T *extends* string ? \\"string\\" : T *extends* number ? \\"number\\" : T *extends* boolean ? \\"boolean\\" : T *extends* undefined ? \\"undefined\\" : \\"object\\""`;

exports[`Types: should compile expanded 'function' type ' 1`] = `"\\\\<T>(arg: T) => T"`;

exports[`Types: should compile expanded 'literal' type' 1`] = `"{ valueA?: number ; valueB?: boolean ; valueX: { valueA: number[] ; valueY: (z: string) => { a: string ; b: string  } ; valueZ: string  } ; valueY: () => string ; valueZ: string  }"`;

exports[`Types: should compile expanded 'objectLiteralType' type' 1`] = `"{ <: string = \\"\\\\<foo>\\"; <foo>: string = \\"foo\\"; \\\\n: string = \\"foo\\"; _: string = \\"foo\\"; prop_with_underscore: string = \\"foo\\"; prop\`with\`backticks: string = \\"foo\\"; prop|with|pipes: string = \\"foo\\"; valueA: number = 100; valueB: boolean = true; valueZ: string = \\"\\\\_foo\\\\_\\"; |: string = \\"foo\\"; ~: string = \\"foo\\"; valueY: (cbParam: \\\\_someCallback\\\\_, unionParam: \\"a\\" \\\\| \\"b\\", \\\\_undercoreParam\\\\_: string) => string ; valueX: { valueA: number[] = [100, 200, 300]; valueZ: string = \\"foo\\"; valueY: (z: string) => { a: string = \\"test\\"; b: string = z }  }  }"`;
=======
exports[`Types: should compile expanded 'function' type ' 1`] = `"<T\\\\>(arg: T) => T"`;

exports[`Types: should compile expanded 'literal' type' 1`] = `"{ valueA?: number ; valueB?: boolean ; valueX: { valueA: number[] ; valueY: (z: string) => { a: string ; b: string  } ; valueZ: string  } ; valueY: () => string ; valueZ: string  }"`;

exports[`Types: should compile expanded 'objectLiteralType' type' 1`] = `"{ <: string = \\"<foo\\\\>\\"; <foo>: string = \\"foo\\"; \\\\n: string = \\"foo\\"; _: string = \\"foo\\"; prop_with_underscore: string = \\"foo\\"; prop\`with\`backticks: string = \\"foo\\"; prop|with|pipes: string = \\"foo\\"; valueA: number = 100; valueB: boolean = true; valueZ: string = \\"\\\\_foo\\\\_\\"; |: string = \\"foo\\"; ~: string = \\"foo\\"; valueY: (cbParam: [\\\\_someCallback\\\\_](../modules/_signatures_.md#_somecallback_), unionParam: \\\\\\"a\\\\\\" \\\\| \\\\\\"b\\\\\\", \\\\_undercoreParam\\\\_: string) => string ; valueX: { valueA: number[] = [100, 200, 300]; valueZ: string = \\"foo\\"; valueY: (z: string) => { a: string = \\"test\\"; b: string = z }  }  }"`;
>>>>>>> master

exports[`Types: should compile intrinsic type' 1`] = `"string"`;

exports[`Types: should compile unionType with object literal type ' 1`] = `"string \\\\| { z: string  }"`;
